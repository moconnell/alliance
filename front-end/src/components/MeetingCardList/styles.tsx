import styled from "styled-components";

export const MeetingCardContainer = styled.div`
  width: auto;
  max-width: 250px;
  display: flex;
  flex-direction: column;
  max-height: 100%;
  overflow-y: auto;
  overflow-x: hidden;
  scrollbar-color: #ff3864 transparent;
  padding: 0 10px 0 10px;

  // For Google Chrome
  &::-webkit-scrollbar {
    border: 1px solid #ff3864;
    border-radius: 26px;
    width: 5px;
  }

  &::-webkit-scrollbar-thumb {
    background: #ff3864;
    border-radius: 26px;
    padding: 0.5px;
  }

  &::-webkit-scrollbar-track {
    background: transparent;
  }

  // For Internet Explorer
  & {
    scrollbar-face-color: #ff3864;
    scrollbar-track-color: transparent;
  }

  &::-moz-scrollbar {
    -moz-border: 1px solid #ff3864;
    border: 1px solid #ff3864;
    border-radius: 26px;
    width: 5px;
  }
  /* Track */

  &::-moz-scrollbar-track {
    background: transparent;
    border: 1px solid #ff3864;
    border-radius: 26px;
  }
  /* Handle */

  &::-moz-scrollbar-thumb {
    background: #ff3864;
    padding: 0.5px;
  }
`;

export const Container = styled.div`
padding: 15px 0;
display: flex;
flex-direction: column;
align-items: center;
height: 400px;
max-height: 400px;
max-width: 250px;
`;

export const SubText = styled.div`
  font-family: Inter;
  font-style: normal;
  font-size: 0.8em;
  line-height: 1em;
  display: flex;
  align-items: center;
  color: #ffffff;
  margin: 1.1em 0 0 0;
`;
