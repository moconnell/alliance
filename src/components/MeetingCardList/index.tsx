import Meeting from "../../types/meeting";
import MeetingCard from "../MeetingCard";
import { Container, MeetingCardContainer, SubText } from "./styles";

interface MeetingCardListProps {
  meetings: Meeting[];
}

const MeetingCardList = ({ meetings }: MeetingCardListProps) => {
  return (
    <Container>
      <MeetingCardContainer>
        {meetings.map((meeting, index) => (
          <MeetingCard key={index + "key"} meeting={meeting} />
        ))}
      </MeetingCardContainer>
      <SubText>Times shown in local time</SubText>
    </Container>
  );
};

export default MeetingCardList;
