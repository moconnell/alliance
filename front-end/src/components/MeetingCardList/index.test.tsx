import { render } from "@testing-library/react";
import MeetingCardList from "./index";

jest.mock("../MeetingCard", () => () => `((MeetingCard))`);

describe("<MeetingCardList />", () => {
  it("should render correctly", async () => {
    const meetings = [
      {
        date: new Date(Date.parse("2022-02-02T20:00.00Z")),
        endDate: new Date(Date.parse("2022-02-02T20:30.00Z")),
        attendee: "Dave Spart",
        description: "Random meeting",
      },
    ];
    const { asFragment } = render(<MeetingCardList meetings={meetings} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
