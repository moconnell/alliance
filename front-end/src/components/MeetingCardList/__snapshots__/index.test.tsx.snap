// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeetingCardList /> should render correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-gueYoa hHsXgJ"
  >
    <div
      class="sc-beqWaB fOYWCI"
    >
      ((MeetingCard))
    </div>
    <div
      class="sc-dmqHEX jpFHOq"
    >
      Times shown in local time
    </div>
  </div>
</DocumentFragment>
`;
